docs: Document PostgreSQL database support for Mautic 7.x (#697)
mautic/user-documentation: docs/getting_started/how_to_install_mautic.rst

Author: *(Line 75)*

PostgreSQL database support reintroduced in [PR #15926](https://github.com/mautic/mautic/pull/15926). Supported range: PostgreSQL 16+, PHP 8.2+. Affects database integration section in docs/getting_started/how_to_install_mautic.rst.

Source: https://github.com/mautic/mautic/pull/15926

> * Your server directory is writable by the Mautic web server.

Author: PR #15926 reintroduces PostgreSQL database support in Mautic 7.x. Supported versions: PostgreSQL 16+, MySQL 8.4+, MariaDB 10.11+. Changes in how_to_install_mautic.rst.

Source: https://github.com/mautic/mautic/pull/15926

> * Your database meets the minimum requirements. Mautic 7 supports MySQL 8.4+, MariaDB 10.11+, and PostgreSQL 16+. Read more in :xref:`Mautic's Requirements`.

Reviewer: This isn't true as Mautic 7.0 and 7.1 won't support PostgreSQL. And I'd rather have 1 source of truth of what databases are supported (link) than several.

Reviewer 2: Then, we'll change it back to the original wording. 👍🏼

> * Your database meets the minimum requirements. Mautic 7 supports MySQL 8.4+, MariaDB 10.11+, and PostgreSQL 16+. Read more in :xref:`Mautic's Requirements`.

Reviewer 2: ```suggestion
* Your database meets the minimum requirements for the supported databases and valid User permissions to access to the database. Read more in :xref:`Mautic's Requirements`.
```

Reviewer 3: @adiati98 This looks good to me overall.
The only real differences for installation/config between MySQL/MariaDB and PostgreSQL are the driver (pdo_mysql vs pdo_pgsql) and the default port (3306 vs 5432). The part about "valid user permissions for the supported databases" feels a bit obvious/redundant now. Most hosting providers expect this as standard.

One small note: since this was built with Mautic 7.x in mind and (as @escopecz mentioned) the docs likely won't merge until 7.2 or later, we should avoid tying it specifically to any early 7.x version.

For setting the database server on the **Mautic Installation-Database Setup** window:

* Select **Database Driver**:

  * ``MySQL PDO`` - for MySQL and MariaDB databases
  * ``PostgreSQL PDO`` - for PostgreSQL databases

* Enter **Database Host**.

* Enter **Database Name**.

* Enter **Database Username**.

> * If desired, you can also enter values for **Database Port** (``3306`` for MySQL/MariaDB, ``5432`` for PostgreSQL), **Database Table Prefix**, **Database Password**, and **Prefix for backup tables**.

Reviewer 2: ```suggestion
* If desired, you can also enter values for **Database Port** - ``3306`` for MySQL/MariaDB, ``5432`` for PostgreSQL, **Database Table Prefix**, **Database Password**, and **Prefix for backup tables**.
```

.. code-block:: php

>      --db_driver=DB_DRIVER                    Database driver (pdo_mysql or pdo_pgsql). [default: "pdo_mysql"]

Reviewer 2: ```suggestion
     --db_driver=DB_DRIVER                    Database driver - "pdo_mysql" or "pdo_pgsql". [default: "pdo_mysql"]
```

      --db_host=DB_HOST                        Database host.

>       --db_port=DB_PORT                        Database port (3306 for MySQL/MariaDB, 5432 for PostgreSQL).

Reviewer 2: ```suggestion
      --db_port=DB_PORT                        Database port - 3306 for MySQL/MariaDB, 5432 for PostgreSQL.
```
